dashboard/src/actions/charts.ts

```diff
@@ -47,35 +84,110 @@ export const errorReadme = createAction("ERROR_README", resolve => {
 });
 
 const allActions = [
-  requestCharts,
-  requestChart,
   errorChart,
-  receiveCharts,
+  errorReadme,
+  reachEnd,
+  receiveChartCategories,
   receiveChartVersions,
-  selectChartVersion,
-  requestDeployedChartVersion,
+  receiveCharts,
+  receiveChartsSearch,
   receiveDeployedChartVersion,
+  requestChart,
+  requestCharts,
+  requestChartsCategories,
+  requestChartsSearch,
+  requestChartsVersions,
+  requestDeployedChartVersion,
   resetChartVersion,
+  resetChartsSearch,
+  resetPaginaton,
+  selectChartVersion,
   selectReadme,
-  errorReadme,
 ];
 
 export type ChartsAction = ActionType<typeof allActions[number]>;
 
-export function fetchCharts(
+export function fetchChartsWithPagination(
   cluster: string,
   namespace: string,
-  repo: string,
-): ThunkAction<Promise<void>, IStoreState, null, ChartsAction> {
+  repos: string,
+  query: string,
+  page: number,
+  size: number,
+  nextPage: number,
```

Comment:
Right,
i) repo**s** is because we can select multiple repos.;
ii) `query` is for searching charts using free text, when `""`it doesn't perform any search;
iii) `page` and `size` is just for the usual pagination. `nextPage` is a workaround for attempting to solve the race condition when requesting